<table>
        <td>
        <td><img src="https://raw.githubusercontent.com/Fabian830348/Bases_Datos/refs/heads/master/logo_ECI.png" width="400"/></td>
        <td>&nbsp;</td>
        <td>
            <h1 style="font-size:200%;color:blue;text-align:center">    <FONT COLOR="blue"> DataFrames </p> Parte 1: Generalidades  </FONT>         </h1></td>         
        <td>
            <tp><p style="font-size:99%;text-align:center">Diplomado Ciencia de Datos </p></tp>
            <tp><p style="font-size:115%;text-align:center">2026-1</p></tp>
            <tp><p style="font-size:115%;text-align:center">Prof. Fabián Sánchez</p></tp>
        </td>
    </tr>
</table>

# <FONT SIZE=5 COLOR="red"> Contenidos del Notebook </FONT>

En este notebook encontrará:

- Concepto de DataFrames
- Construir un DataFrame con pandas
- Carga de Datos
  + Desde drive
  + Desde el Github
- Exploración de datos
  + *.head()*
  + *.tail()*
  + *.shape*
  + *.info()*
  + *.columns*
  + *.describe()*
- Tablas de Frecuencias
  + *.value_counts()*
  + *.crosstab()*
- Estadísticas básicas
  + *.mean()*
  + *.median()*
  + *.mode()*
  + *.quantile()*


# <FONT SIZE=5 COLOR="green"> 1. Definición de DataFrame en Python </FONT>

- En este notebook trabajaremos uno de los objetos más importante cuando manipulamos y procesamos datos, los DataFrames.

- Recordemos que existen dos grupos grandes de datos que se denominan:

  - **Datos estructurados**: son aquellos que se pueden representar en un arreglo bidimensional: *fila-columna*. Es decir, las clásicas hojas de cálculo de excel.
  - **Datos no estructurados**: son los que no podemos representar en forma estructurada *fila-columna*. Por ejemplo, archivos de vídeo, audio, texto, imágenes, etc.


 De acuerdo con lo anterior definimos:

 ***Dataframe***: son arreglos o estructuras bidimensionales compuestos por fila y por columnas, y cuyos datos pueden ser de diferentes tipos: números, cadenas de texto, booleanos, etc.

   - Un dataframe tiene una forma matricial donde se identifican filas y columnas. Soporta diferentes tipos de datos.
   - Un dataframe está compuesto por vectores (columnas) cada una de las cuales describe una atributo o característica (variable) de una población.
   - La librería más usada para manipulación de dataframes se denomina ***pandas*** en Python. Si trabajas en *RStudio* es ***dplyr***

<br>
<center><img src="https://github.com/Fabian830348/cursos/blob/master/intro/DATAFRAME.png?raw=true" alt="centered image" width="600" height="350"></center> <center><figcaption> <FONT SIZE=1 COLOR="black"> Fuente: Elaboración propia  </FONT> <figcaption></center>




Iniciaremos importando la librería *pandas*

In [ ]:
import pandas as pd
import numpy as np

# <FONT SIZE=5 COLOR="green"> 2. Generación de un DataFrame </FONT>

En esta sección revisaremos cómo se genera un dataframe a partir de:

- Listas
- Un diccionario

para generar un dataframe usamos la instrucción de pandas

$$pd.DataFrame()$$

## <FONT SIZE=4 COLOR="magenta"> 2.1 A partir de una Lista </FONT>

In [ ]:
lista1 = ["Fabian" , 30 , 164 , "Doctorado"]
lista2 = ["Camilo" , 24 , 174 , "Maestría"]
lista3 = ["Ana María" , 20 , 180 , "Profesional"]
lista4 = ["Andrea" , 36 , 154 , "Doctorado"]
lista5 = ["Mariana" , 45 , 164 , "Doctorado"]

pd.DataFrame([lista1, lista2, lista3, lista4, lista5])

observe que enumera las columnas pero no tiene nombres.

Ahora, vamos a colocar nombre a las columnas

In [ ]:
lista1 = ["Fabian" , 30 , 164 , "Doctorado"]
lista2 = ["Camilo" , 24 , 174 , "Maestría"]
lista3 = ["Ana María" , 20 , 180 , "Profesional"]
lista4 = ["Andrea" , 36 , 154 , "Doctorado"]
lista5 = ["Mariana" , 45 , 164 , "Doctorado"]

df1 = pd.DataFrame([lista1, lista2, lista3, lista4, lista5],                          # datos
              columns = ["Nombre", "Edad", "Estatura(cm)", "Nivel_Educativo"])        # variables

In [ ]:
df1

## <FONT SIZE=4 COLOR="magenta"> 2.2 A partir de un diccionario </FONT>

Para construir el DataFrame a partir de un diccionario se usa

$$pd.DataFrame(diccionario)$$

Si se tienen las listas que representan las columnas, el dataframe se construye de la siguiente forma

In [ ]:
lista1 = [30,40,50,60]
lista2 = ["Audi", "BMW", "Renault", "ByD"]
lista3 = [1.76, 1.50, 1.60, 1.80]
lista4 = ["Carlos", "Juan", "Hugo", "Ana"]

diccionario = {"Edad" : lista1,
               "Marca_Auto" : lista2,
               "Estatura(mts)" : lista3,
               "Nombre" : lista4}

df2 =pd.DataFrame(diccionario)
df2

Podemos colocar el diccionario directamente dentro de la expresión DataFrame

In [ ]:
diccionario = {"Nombre" : ["Hugo", "Andres", "Camilo", "Ana" ],
               "Edad" : [70, 40, 24, 10],
               "Estatura" : [180, 170, 150, 176],
               "Nivel_Educativo" : ["Doctorado", "Profesional", "Maestría","Maestría"]}

df3 = pd.DataFrame(diccionario)
df3

In [ ]:
# ejemplo
nombres = ["Hugo", "Andres", "Camilo", "Ana"]
edad =  [70, 40, 24, 10]
pd.DataFrame({"nombres" : nombres , "edad" : edad})


Es importante conocer como generar un DataFrame desde una lista o diccionario, sin embargo lo más importante es saber cómo cargar datos de diferentes fuentes.

# <FONT SIZE=5 COLOR="Green"> 3. Carga de Datos </FONT></center>

Vamos a cargar datos (dataframe) de dos formas diferentes:

  - Desde ***googledrive()***
  - Desde un repositorio externo. Generalmente GitHub.

## <FONT SIZE=4 COLOR="magenta"> 3.1 Usando Google-Drive </FONT></center>

- Para conectarse al drive usamos la siguiente instrucción

In [ ]:
# instrucción para conectarse con drive
from google.colab import drive
drive.mount('/content/drive')

Siempre que trabajemos con dataframes es importante cargar la librería pandas.

In [ ]:
#para importar la librería pandas.
import pandas as pd

Para cargar los archivos buscamos la ubicación de los mismos y la ruta en la parte izquierda del notebook

In [14]:
ruta_drive = "/content/drive/MyDrive/11. Diplomado Ciencia de Datos Grupo 2 (Mayo-Junio-2026)/DATOS/apartamentos.csv"

Como el archivo está en formato *.csv*, que es un archivo separado por coma usamos la función

$$pd.read\_csv(link)$$

In [15]:
aptos = pd.read_csv(ruta_drive)
aptos

,Unnamed: 0,precio,mt2,ubicacion,estrato,alcobas,banos,balcon,parqueadero,administracion,avaluo,terminado
0,1,79.0,43.16,norte,3,3,1,si,si,0.050,14.923002,no
1,2,93.0,56.92,norte,2,2,1,si,si,0.069,27.000000,si
2,3,100.0,66.40,norte,3,2,2,no,no,0.000,15.738427,no
3,4,123.0,61.85,norte,2,3,2,si,si,0.130,27.000000,no
4,5,135.0,89.80,norte,4,3,2,si,no,0.000,39.567000,si
...,...,...,...,...,...,...,...,...,...,...,...,...
689,690,498.0,180.00,laureles,5,4,3,si,si,0.570,233.784000,si
690,691,508.0,209.00,laureles,6,4,3,si,si,0.450,258.850000,si
691,692,550.0,149.00,laureles,6,3,4,no,si,0.747,326.157000,si
692,693,630.0,179.00,laureles,5,3,3,si,si,0.340,324.114000,si


Si el archivo está en formato *.xlsx*, usamos la función

$$pd.read\_excel(link)$$

In [16]:
# traer el conjunto de datos hepatitis
hepatitis = pd.read_excel("/content/drive/MyDrive/11. Diplomado Ciencia de Datos Grupo 2 (Mayo-Junio-2026)/DATOS/hepatitis.xlsx")
hepatitis

,Clase,Edad,Sexo,Esteroides,Antivirales,Fatiga,Malestar,Anorexia,Higado Grande,Cirrosis,Bazo palpable,Arañas,Ascites,Varices,Billirubina,Fosfato,SGOT,Albumina,Protina,Histologia
0,Vivo,30,Mujer,No,Sí,Sí,Sí,Sí,No,Sí,Sí,Sí,Sí,Sí,1.0,85,18,4.0,0,No
1,Vivo,50,Hombre,No,Sí,No,Sí,Sí,No,Sí,Sí,Sí,Sí,Sí,0.9,135,42,3.5,0,No
2,Vivo,78,Hombre,Sí,Sí,No,Sí,Sí,Sí,Sí,Sí,Sí,Sí,Sí,0.7,96,32,4.0,0,No
3,Vivo,31,Hombre,,No,Sí,Sí,Sí,Sí,Sí,Sí,Sí,Sí,Sí,0.7,46,52,4.0,80,No
4,Vivo,34,Hombre,Sí,Sí,Sí,Sí,Sí,Sí,Sí,Sí,Sí,Sí,Sí,1.0,0,200,4.0,0,No
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
147,Muerto,46,Hombre,Sí,Sí,No,No,No,Sí,Sí,Sí,No,No,No,7.6,0,242,3.3,50,Sí
148,Vivo,44,Hombre,Sí,Sí,No,Sí,Sí,Sí,No,Sí,Sí,Sí,Sí,0.9,126,142,4.3,0,Sí
149,Vivo,61,Hombre,No,Sí,No,No,Sí,No,No,Sí,No,Sí,Sí,0.8,75,20,4.1,0,Sí
150,Vivo,53,Mujer,No,Sí,No,Sí,Sí,Sí,Sí,No,No,Sí,No,1.5,81,19,4.1,48,Sí


Si el archivo está en formato *.json*, usamos la función

$$pd.read\_json(link)$$

In [17]:
# traer el conjunto de datos .json
espectro = pd.read_json("/content/drive/MyDrive/11. Diplomado Ciencia de Datos Grupo 2 (Mayo-Junio-2026)/DATOS/espectro.json")
espectro

,nombre_sonda,ciudad_departamento,fecha_medicion,nivel_exposicion_campos,limite_restrictivo,porcentaje_exposicion,geocoded_column
0,Bogota_BOSA_2,Bogota_BOGOTA,2020-06-30T23:54:00.000,1.11,28,3.96,"{'type': 'Point', 'coordinates': [-74.1688982,..."
1,Santa_Marta_TERMINAL,Santa_Marta_MAGDALENA,2020-06-30T23:54:00.000,2.14,28,7.64,"{'type': 'Point', 'coordinates': [-74.18208333..."
2,Zipaquira_ZIPA_CONSTRUCTORA,Zipaquira_CUNDINAMARCA,2020-06-30T23:54:00.000,1.53,28,5.46,"{'type': 'Point', 'coordinates': [-74.00063, 5..."
3,Armenia_CENTRO,Armenia_QUINDIO,2020-06-30T23:54:00.000,1.51,28,5.39,"{'type': 'Point', 'coordinates': [-75.67364, 4..."
4,Pereira_CENTENARIO,Pereira_RISARALDA,2020-06-30T23:54:00.000,0.00,28,0.00,"{'type': 'Point', 'coordinates': [-75.698193, ..."
...,...,...,...,...,...,...,...
995,Soacha_CENTRO,Soacha_CUNDINAMARCA,2020-06-30T22:06:00.000,1.93,28,6.89,"{'type': 'Point', 'coordinates': [-74.220488, ..."
996,Santa_Marta_TERMINAL,Santa_Marta_MAGDALENA,2020-06-30T22:06:00.000,2.06,28,7.36,"{'type': 'Point', 'coordinates': [-74.18208333..."
997,Barranquilla_SAN_FELIPE,Barranquilla_ATLANTICO,2020-06-30T22:06:00.000,2.11,28,7.54,"{'type': 'Point', 'coordinates': [-74.80913888..."
998,Pereira_BERLIN,Pereira_RISARALDA,2020-06-30T22:06:00.000,1.80,28,6.43,"{'type': 'Point', 'coordinates': [-75.68085, 4..."


## <FONT SIZE=5 COLOR="magenta"> 3.2 Usando Repositorio Externo url </FONT></center>

Para importar datos desde un repositorio externo, debemos conocer la *url*.

https://archive.ics.uci.edu/datasets?skip=10&take=10&sort=desc&orderBy=NumHits&search=

https://github.com/stedy/Machine-Learning-with-R-datasets

In [18]:
# conjunto de datos crédito .csv
credito = pd.read_csv("https://raw.githubusercontent.com/stedy/Machine-Learning-with-R-datasets/master/credit.csv")
credito

,checking_balance,months_loan_duration,credit_history,purpose,amount,savings_balance,employment_length,installment_rate,personal_status,other_debtors,...,property,age,installment_plan,housing,existing_credits,default,dependents,telephone,foreign_worker,job
0,< 0 DM,6,critical,radio/tv,1169,unknown,> 7 yrs,4,single male,none,...,real estate,67,none,own,2,1,1,yes,yes,skilled employee
1,1 - 200 DM,48,repaid,radio/tv,5951,< 100 DM,1 - 4 yrs,2,female,none,...,real estate,22,none,own,1,2,1,none,yes,skilled employee
2,unknown,12,critical,education,2096,< 100 DM,4 - 7 yrs,2,single male,none,...,real estate,49,none,own,1,1,2,none,yes,unskilled resident
3,< 0 DM,42,repaid,furniture,7882,< 100 DM,4 - 7 yrs,2,single male,guarantor,...,building society savings,45,none,for free,1,1,2,none,yes,skilled employee
4,< 0 DM,24,delayed,car (new),4870,< 100 DM,1 - 4 yrs,3,single male,none,...,unknown/none,53,none,for free,2,2,2,none,yes,skilled employee
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,unknown,12,repaid,furniture,1736,< 100 DM,4 - 7 yrs,3,female,none,...,real estate,31,none,own,1,1,1,none,yes,unskilled resident
996,< 0 DM,30,repaid,car (used),3857,< 100 DM,1 - 4 yrs,4,divorced male,none,...,building society savings,40,none,own,1,1,1,yes,yes,mangement self-employed
997,unknown,12,repaid,radio/tv,804,< 100 DM,> 7 yrs,4,single male,none,...,other,38,none,own,1,1,1,none,yes,skilled employee
998,< 0 DM,45,repaid,radio/tv,1845,< 100 DM,1 - 4 yrs,4,single male,none,...,unknown/none,23,none,for free,1,2,1,yes,yes,skilled employee


In [19]:
# conjunto de datos diabetes .csv
diabetes = pd.read_csv("https://raw.githubusercontent.com/Fabian830348/Bases_Datos/refs/heads/master/diabetes.csv ")
diabetes

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,6,148,72,35,0,33.6,0.627,50,1
1,1,85,66,29,0,26.6,0.351,31,0
2,8,183,64,0,0,23.3,0.672,32,1
3,1,89,66,23,94,28.1,0.167,21,0
4,0,137,40,35,168,43.1,2.288,33,1
...,...,...,...,...,...,...,...,...,...
763,10,101,76,48,180,32.9,0.171,63,0
764,2,122,70,27,0,36.8,0.340,27,0
765,5,121,72,23,112,26.2,0.245,30,0
766,1,126,60,0,0,30.1,0.349,47,1


In [20]:
# conjunto de datos titanic en .json
titanic = pd.read_json("https://raw.githubusercontent.com/joanby/python-ml-course/refs/heads/master/datasets/titanic/titanic_custom.json")
titanic

,pclass,survived,name,sex,age,sibsp,parch,ticket,fare,cabin,embarked,boat,body,home.dest
0,1,1,"Allen, Miss. Elisabeth Walton",female,29.0000,0,0,24160,211.3375,B5,S,2,NaN,"St Louis, MO"
1,1,1,"Allison, Master. Hudson Trevor",male,0.9167,1,2,113781,151.5500,C22 C26,S,11,NaN,"Montreal, PQ / Chesterville, ON"
2,1,0,"Allison, Miss. Helen Loraine",female,2.0000,1,2,113781,151.5500,C22 C26,S,None,NaN,"Montreal, PQ / Chesterville, ON"
3,1,0,"Allison, Mr. Hudson Joshua Creighton",male,30.0000,1,2,113781,151.5500,C22 C26,S,None,135.0,"Montreal, PQ / Chesterville, ON"
4,1,0,"Allison, Mrs. Hudson J C (Bessie Waldo Daniels)",female,25.0000,1,2,113781,151.5500,C22 C26,S,None,NaN,"Montreal, PQ / Chesterville, ON"
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1304,3,0,"Zabour, Miss. Hileni",female,14.5000,1,0,2665,14.4542,None,C,None,328.0,None
1305,3,0,"Zabour, Miss. Thamine",female,NaN,1,0,2665,14.4542,None,C,None,NaN,None
1306,3,0,"Zakarian, Mr. Mapriededer",male,26.5000,0,0,2656,7.2250,None,C,None,304.0,None
1307,3,0,"Zakarian, Mr. Ortin",male,27.0000,0,0,2670,7.2250,None,C,None,NaN,None


# <FONT SIZE=5 COLOR="green"> 4. Instrucciones básicas de exploración de DataFrames</FONT>

Algunas instrucciones básicas de un dataframe:

**df.info()** : Devuelve información (número de filas, número de columnas, índices, tipo de las columnas y memoria usado) sobre el DataFrame df.

**df.shape** : Devuelve una pareja con el número de filas y columnas del DataFrame df.

**df.size** : Devuelve el número de elementos del DataFrame.

**df.columns** : Devuelve una lista con los nombres de las columnas del DataFrame df.

**df.index** : Devuelve una lista con los nombres de las filas del DataFrame df.

**df.dtypes** : Devuelve una serie con los tipos de datos de las columnas del DataFrame df.

**df.head(n)** : Devuelve las n primeras filas del DataFrame df.

**df.tail(n)** : Devuelve las n últimas filas del DataFrame df.

**df.append({})** : para agregar una nueva fila al dataframe

A continuación revisaremos algunas de ellas con el conjunto de datos apartamentos

Para lo que sigue vamos a usar el conjunto **apartamentos**

In [21]:
# conjunto de datos aptos.head()
aptos

,Unnamed: 0,precio,mt2,ubicacion,estrato,alcobas,banos,balcon,parqueadero,administracion,avaluo,terminado
0,1,79.0,43.16,norte,3,3,1,si,si,0.050,14.923002,no
1,2,93.0,56.92,norte,2,2,1,si,si,0.069,27.000000,si
2,3,100.0,66.40,norte,3,2,2,no,no,0.000,15.738427,no
3,4,123.0,61.85,norte,2,3,2,si,si,0.130,27.000000,no
4,5,135.0,89.80,norte,4,3,2,si,no,0.000,39.567000,si
...,...,...,...,...,...,...,...,...,...,...,...,...
689,690,498.0,180.00,laureles,5,4,3,si,si,0.570,233.784000,si
690,691,508.0,209.00,laureles,6,4,3,si,si,0.450,258.850000,si
691,692,550.0,149.00,laureles,6,3,4,no,si,0.747,326.157000,si
692,693,630.0,179.00,laureles,5,3,3,si,si,0.340,324.114000,si


1. Usamos la instrucción **head()** para explorar la cabeza de los datos

- Podemos indicar cuántos datos queremos visualizar simplemente con $head(n)$, sonde $n$ indica el número de registros que queremos ver.

In [22]:
# para visualizar los primeros datos.
aptos.head(4)

,Unnamed: 0,precio,mt2,ubicacion,estrato,alcobas,banos,balcon,parqueadero,administracion,avaluo,terminado
0,1,79.0,43.16,norte,3,3,1,si,si,0.050,14.923002,no
1,2,93.0,56.92,norte,2,2,1,si,si,0.069,27.000000,si
2,3,100.0,66.40,norte,3,2,2,no,no,0.000,15.738427,no
3,4,123.0,61.85,norte,2,3,2,si,si,0.130,27.000000,no


2. Usamos la instrucción **tail()** para explorar los últimos datos

In [23]:
# para visualizar los últimos datos. (n=6)
aptos.tail(6)

,Unnamed: 0,precio,mt2,ubicacion,estrato,alcobas,banos,balcon,parqueadero,administracion,avaluo,terminado
688,689,470.0,248.0,laureles,5,3,2,no,si,0.4809,345.376,si
689,690,498.0,180.0,laureles,5,4,3,si,si,0.5700,233.784,si
690,691,508.0,209.0,laureles,6,4,3,si,si,0.4500,258.850,si
691,692,550.0,149.0,laureles,6,3,4,no,si,0.7470,326.157,si
692,693,630.0,179.0,laureles,5,3,3,si,si,0.3400,324.114,si
693,694,1350.0,494.0,laureles,5,3,4,si,si,1.6000,320.000,si


3. Con la instrucción **shape** miramos el tamaño de nuestros datos

In [24]:
# tamaño de los datos
aptos.shape

(694, 12)

el conjunto tiene 694 filas y 12 columnas.

4. Para explorar nombre de las variables o columnas **columns**

In [25]:
# columnas del conjunto de datos
aptos.columns

Index(['Unnamed: 0', 'precio', 'mt2', 'ubicacion', 'estrato', 'alcobas',
       'banos', 'balcon', 'parqueadero', 'administracion', 'avaluo',
       'terminado'],
      dtype='object')

5. **info()** es un resumen de las variables y su clasificación. Además, podemos identificar datos faltantes (NA) en cada una de ellas.

In [26]:
# información de los datos
aptos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 694 entries, 0 to 693
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      694 non-null    int64  
 1   precio          694 non-null    float64
 2   mt2             694 non-null    float64
 3   ubicacion       694 non-null    object 
 4   estrato         694 non-null    int64  
 5   alcobas         694 non-null    int64  
 6   banos           694 non-null    int64  
 7   balcon          694 non-null    object 
 8   parqueadero     694 non-null    object 
 9   administracion  694 non-null    float64
 10  avaluo          694 non-null    float64
 11  terminado       694 non-null    object 
dtypes: float64(4), int64(4), object(4)
memory usage: 65.2+ KB


Es importante verificar que las variables del conjunto de datos hayan sido cargadas en el tipo adecuado.

6. Para revisar las estadísticas generales de las variables cuantitativas usamos **describe()**

In [27]:
# estadísticas descriptivas
aptos.describe()

,Unnamed: 0,precio,mt2,estrato,alcobas,banos,administracion,avaluo
count,694.000000,694.000000,694.000000,694.000000,694.000000,694.000000,694.000000,694.000000
mean,347.500000,317.762739,120.943818,4.651297,2.910663,2.289625,0.263556,182.636593
std,200.484829,247.614914,74.695493,1.208205,0.763962,0.795718,0.251586,169.082384
min,1.000000,25.000000,26.000000,2.000000,1.000000,1.000000,0.000000,0.149000
25%,174.250000,160.000000,71.000000,4.000000,3.000000,2.000000,0.090000,73.000000
50%,347.500000,245.000000,97.500000,5.000000,3.000000,2.000000,0.201000,131.431500
75%,520.750000,380.000000,141.750000,6.000000,3.000000,3.000000,0.375000,234.051000
max,694.000000,1700.000000,500.000000,6.000000,14.000000,6.000000,2.280000,1540.620000


In [28]:
# si queremos las estadísticas (frecuencias) de las variables cualitativas
aptos.describe(include='object')

,ubicacion,balcon,parqueadero,terminado
count,694,694,694,694
unique,7,2,2,2
top,poblado,si,si,si
freq,268,509,568,592


7. Para acceder a una o varias variable del data.frame podemos usar la instrucción

*datos.["nombre de la variable"]*

veamos algunos ejemplos

In [29]:
# podemos acceder a una o más variables del data.frame data[]
aptos["precio"]

,precio
0,79.0
1,93.0
2,100.0
3,123.0
4,135.0
...,...
689,498.0
690,508.0
691,550.0
692,630.0


In [30]:
# acceder a varias columnas
aptos[["ubicacion","estrato","banos"]]

,ubicacion,estrato,banos
0,norte,3,1
1,norte,2,1
2,norte,3,2
3,norte,2,2
4,norte,4,2
...,...,...,...
689,laureles,5,3
690,laureles,6,3
691,laureles,6,4
692,laureles,5,3


In [31]:
# valores únicos de una variable
aptos.ubicacion.unique()

array(['norte', 'occidente', 'poblado', 'aburra sur', 'belen guayabal',
       'centro', 'laureles'], dtype=object)

In [32]:
# podemos revisar unas estadísticas básicas de una variable numérica con .describe()
aptos["precio"].describe()

,precio
count,694.000000
mean,317.762739
std,247.614914
min,25.000000
25%,160.000000
50%,245.000000
75%,380.000000
max,1700.000000


In [33]:
aptos[["precio", "mt2"]]

,precio,mt2
0,79.0,43.16
1,93.0,56.92
2,100.0,66.40
3,123.0,61.85
4,135.0,89.80
...,...,...
689,498.0,180.00
690,508.0,209.00
691,550.0,149.00
692,630.0,179.00


También se puede acceder al nombre de una variable asi:
$$datos.variable$$
Siempre y cuando el nombre de la variable no tenga espacios o caracteres especiales.

In [34]:
aptos.precio

,precio
0,79.0
1,93.0
2,100.0
3,123.0
4,135.0
...,...
689,498.0
690,508.0
691,550.0
692,630.0


# <FONT SIZE=5 COLOR="green"> 5. Tabla de frecuencias </FONT>

Para hacer el conteo de las categorías o clases de una variables podemos usar la instrucción ***.value_counts***

Ejemplo: Para hacer una tabla de frecuencias de la variable *ubicacion*

In [35]:
aptos["ubicacion"].value_counts()

,count
ubicacion,
poblado,268
aburra sur,169
laureles,73
occidente,69
belen guayabal,67
centro,38
norte,10


In [36]:
# conteo de la variable terminado-aptos
aptos["terminado"].value_counts()

,count
terminado,
si,592
no,102


Si queremos una tabla cruzada entre dos variables categóricas usamos la instrucción:

$$pd.crosstab()$$

In [37]:
# tabla cruzada de ubicación y estrato
pd.crosstab(aptos.ubicacion , aptos.estrato)

estrato,2,3,4,5,6
ubicacion,,,,,
aburra sur,3,73,57,35,1
belen guayabal,1,30,21,15,0
centro,0,16,21,1,0
laureles,0,0,14,56,3
norte,2,5,3,0,0
occidente,2,37,15,15,0
poblado,0,0,7,23,238


In [38]:
# tabla de frecuencia (contingencia) ubicacion y parqueadero
# las dos instrucciones adicionales sirven para totalizar
pd.crosstab(aptos["ubicacion"],                 # variable 1
            aptos["parqueadero"],               # variable 2
            margins= True,                      # muestra el total
            margins_name= "SubTotal")           # nombre del total

parqueadero,no,si,SubTotal
ubicacion,,,
aburra sur,39,130,169
belen guayabal,31,36,67
centro,28,10,38
laureles,3,70,73
norte,3,7,10
occidente,20,49,69
poblado,2,266,268
SubTotal,126,568,694


In [39]:
# tabla de contingencia de tres variables cruzadas
pd.crosstab(aptos["ubicacion"], [aptos["balcon"], aptos["estrato"]])

balcon         no                 si                 
estrato         2   3   4   5   6  2   3   4   5    6
ubicacion                                            
aburra sur      2  26   6   4   0  1  47  51  31    1
belen guayabal  0   6   5   1   0  1  24  16  14    0
centro          0   8  10   0   0  0   8  11   1    0
laureles        0   0   5  18   2  0   0   9  38    1
norte           0   2   0   0   0  2   3   3   0    0
occidente       1  17   7   3   0  1  20   8  12    0
poblado         0   0   2   9  51  0   0   5  14  187

# <FONT SIZE=5 COLOR="green"> 6. Algunas funciones importantes </FONT>

A continuación revisaremos algunas estadísticas básicas

- Media
- Mediana
- Moda
- Cuantiles

1. Media o promedio: **.mean()**

- Es el "equilibrio" de los datos.
- Se usa para variables cuantitativas.
- Siempre está entre el mínimo y máximo de los datos.
- Es sensible a datos atípicos. Por está razón se debe tener cuidado con su interpretación.

In [40]:
aptos.precio.mean()

np.float64(317.7627391930836)

2. Mediana: **.median()**

- Es el valor que divide los datos en dos partes iguales (50%-50%)
- Aplica para variables cuantitativas
- Puede o no estar en el conjunto de datos. Depende si se tiene un número par o impar de elementos.
- No se ve afectada por la existencia de datos atípicos. Por está razón en ocaciones se usa para imputación.

In [41]:
aptos.precio.median()

245.0

3. Moda: **.mode()**

- Es el valor que más se repite. Es decir el que tiene mayor frecuencia
- Aplica para variables cualitativas y cuantitativas
- UN conjunto puede no tener o tener múltiples modas.

In [42]:
aptos.ubicacion.mode()

,ubicacion
0,poblado


4. Máximo y mínimo del conjunto: **.max** y **.min**

In [43]:
aptos.precio.max()

1700.0

In [44]:
aptos.precio.min()

25.0

5. Cuartiles: $Q_1$, $Q_2$ y $Q_3$

- Se calculan para variables cuantitativas
- $Q_1$: el valor que acumula el 25% de los datos
- $Q_2$: es el valor que acumula el 50% de los datos. En ese orden de ideas es la mediana.
- $Q_3$ es el valor que acumula el 75% de los datos.
- Para calcula en Python estos valores y en general los cuantiles usamos la librería *numpy* y la instrucción:

$$ perc\_{n} = np.percentile(variable, n) $$

Veamos algunos ejemplos con la variable *precio* de *aptos*.

In [45]:
import numpy as np
print(np.percentile(aptos.precio,25))
print(np.percentile(aptos.precio,50))
print(np.percentile(aptos.precio,75))
print(np.percentile(aptos.precio,80))

160.0
245.0
380.0
421.9999999999999
